static/js/search_util.ts

```diff
@@ -1,12 +1,17 @@
-export function get_search_terms(input) {
+export function get_search_terms(input: string): string[] {
     const search_terms = input
         .toLowerCase()
         .split(",")
         .map((s) => s.trim());
     return search_terms;
 }
 
-export function vanilla_match(opts) {
+interface VanillaMatchOpts {
+    val: string;
+    search_terms: string[];
+}
+
+export function vanilla_match(opts: VanillaMatchOpts): boolean {
```

Comment:
I suspect we’ll generally be happier inlining types that are so specific to a single function’s calling convention:

```ts
export function vanilla_match(opts: {
    val: string;
    search_terms: string[];
}): boolean {
```

so that

* we can look at a function and see immediately how to call it, without referring to an auxiliary type;
* we don’t have to spend effort coming up with names for these auxiliary types;
* we reduce the activation energy for future refactors that change the calling convention.

Reply:
Done.